Implement syncer (#367)
kubernetes-sigs/kube-scheduler-simulator: simulator/docs/import-cluster-resources.md

### [Beta] Import your real cluster's resources

> There are two ways to import resources from your cluster.

Reviewer: It would be better to mention that these two methods can be used simultaneously

### [Beta] Import your real cluster's resources

There are two ways to import resources from your cluster. These methods can be used simultaneously.
- Import resources from your cluster once when initializing the simulator.
- Keep importing resources from your cluster.

#### Import resources once when initializing the simulator

To use this, you need to follow these two steps in the simulator configuration:
- Set `true` to `externalImportEnabled`.
- Set the path of the kubeconfig file for the your cluster to `KubeConfig`. 
  - This feature only requires the read permission for resources.

```yaml
externalImportEnabled: true
kubeConfig: "/path/to/your-cluster-kubeconfig"
```

#### Keep importing resources

To use this, you need to follow these two steps in the scheduler configuration:

> - Set `true` to `externalImportSynced`. 

Reviewer: ```suggestion
- Set `true` to `resourceSyncEnabled`. 
```

- Set the path of the kubeconfig file for the your cluster to `KubeConfig`. 
  - This feature only requires the read permission for resources.

```yaml

> externalSchedulerSynced: true

Reviewer: ```suggestion
resourceSyncEnabled: true
```

kubeConfig: "/path/to/your-cluster-kubeconfig"
```

> [!NOTE]

> > When you enable `resourceSyncEnabled`, deleting resources of the simulator cluster would cause a problem of syncing. Please avoid it except when you are debugging the simulator. When you delete a resource for debugging, make sure you reboot the simulator and the fake source cluster.

Reviewer 2: ```suggestion
> When you enable `resourceSyncEnabled`, adding/updating/deleting resources directly in the simulator cluster could cause a problem of syncing. 
> You can do them for debugging etc purposes though, make sure you reboot the simulator and the fake source cluster afterward.
```

> > When you enable `externalImportSynced`, deleting resources of the simulator cluster would cause a problem of syncing. Please avoid it except when you are debugging the simulator. When you delete a resource for debugging, make sure you reboot the simulator and the fake source cluster.

Reviewer: ```suggestion
> When you enable `resourceSyncEnabled`, deleting resources of the simulator cluster would cause a problem of syncing. Please avoid it except when you are debugging the simulator. When you delete a resource for debugging, make sure you reboot the simulator and the fake source cluster.
```
